controllers/metal3.io/hostfirmwarecomponents_controller.go

```diff
@@ -196,35 +191,45 @@ func (r *HostFirmwareComponentsReconciler) updateHostFirmware(info *rhfcInfo, co
 	reason := reasonValidComponent
 	generation := info.hfc.GetGeneration()
 
+	newStatus := info.hfc.Status.DeepCopy()
+	// change the Components in Status only if they differ
+	if !reflect.DeepEqual(info.hfc.Status.Components, components) {
+		newStatus.Components = make([]metal3api.FirmwareComponentStatus, len(components))
+		for i := range info.hfc.Status.Components {
```

Comment:
I think you want to loop over the `components` variable instead of the old status? If the status is empty, the loop never runs.

Reply:
right, makes sense!